### ISSUE TITLE
Help in adding new feature

### ISSUE THREAD

**Issue by kapurm17** [sukeesh/Jarvis#377]
Hey! 
I have written a python script for a new feature for the project .
Can someone guide me  with adding this new feature in the script  Which files have to be edited? 

P.S - I'm a beginner and this is my first project , so please spare me if this is a queston with an obvious answer.
Thanks !

**Comment by pnhofmann** [comment]
Hi!

> Can someone guide me with adding this new feature in the script Which files have to be edited?

Actually: None ;).

As descried [here](https://github.com/sukeesh/Jarvis#how-to-write-new-commands-plugins-to-extend-jarvis-functionality) just place your script in ``jarviscli/plugins``. Then add to the top:

```
from plugin import plugin

@plugin()
def yourpluginname(jarvis, s):
    """
    Documentation of your plugin
    """"
    YourCode
```

Try it out, run Jarvis and type ``yourpluginname`` - your feature should be called!

Have Fun and don't hesitate asking if something does not work as expected!

**Comment by pnhofmann** [comment]
Email, Nice! Looks good and really useful!

I'm perfectly fine about your code. Just about formality:

We use [flake8](https://pypi.org/project/flake8/) for code style checking checking.

Do something like:
```
source env/bin/activate
./test.sh
# or to only run flake and no unit tests
flake8 --select E,W --max-line-length=140 --ignore E722 jarviscli/
```
Ignore the 3 errors in CmdInterpreter and one in weatherIn - will handle those...

Still some issues are reported in gmail.py - e.g. never use tabs with python, add a space after inline comment with # and don't add spaces at end of line.

All in all just minor stuff - but following those rules helps getting a clean and sustainable project.

> Can i have the push access?

Generally on Github this is done bit different. You generally always fork and upload your changes to your own repo (like you already did).
Then (if flake8 reports no errors ;) ) press a button "Create pull request" - and the "pull request" basically notify us, that you want to submit new code.

I (or other Collaborator/Owner of this Repository) will have the chance to review your changes and merge afterwards.

More information:

https://help.github.com/articles/about-pull-requests/

**Comment by kapurm17** [comment]
Thanks!
I used tabs not to indent the code , but for mere spacing, and i realized that shouldn't be done either.
I've done the flake8 and it returned no error so I created a pull request. :)

**Comment by pnhofmann** [comment]
Thanks! Merged #378.
